pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -578,8 +578,13 @@ TEST(BasicEndToEndTest, testSinglePartitionRoutingPolicy)
         consumer.receive(m);
         consumer.acknowledgeCumulative(m);
     }
+
+    Promise<bool, Result> closePromise;
+    Result r;
     consumer.close();
-    producer.closeAsync(0);
+    producer.closeAsync(WaitForCallback(closePromise));
```

Comment:
Couldn't you just use `producer.close()` here. That should be the exact same implementation internally.